locale/en/docs/guides/backpressuring-in-streams.md

```diff
@@ -0,0 +1,264 @@
+implement our own custom streams.
+
+## Lifecycle of `.pipe()`
+
+To achieve a better understanding of backpressure, here is a flow-chart on the 
+lifecycle of a [`Readable`][] stream being [piped][] into a [`Writable`][] 
+stream: 
+
+```javascript
++===============+
+|   Your_Data   |
++=======+=======+
+        |
++-------v-----------+          +-------------------+         +=================+
+|  Readable Stream  |          |  Writable Stream  +--------->  .write(chunk)  |
++-------+-----------+          +---------^---------+         +=======+=========+
+        |                                |                           |
+        |     +======================+   |        +------------------v---------+
+        +----->  .pipe(destination)  >---+        |    Is this chunk too big?  |
+              +==^=======^========^==+            |    Is the queue busy?      |
+                 ^       ^        ^               +----------+-------------+---+
+                 |       |        |                          |             |
+                 |       |        |  > if (!chunk)           |             |
+                 ^       |        |      emit .end();        |             |
+                 ^       ^        |  > else                  |             |
+                 |       ^        |      emit .write();  +---v---+     +---v---+
+                 |       |        ^----^-----------------<  No   |     |  Yes  |
+                 ^       |                               +-------+     +---v---+
+                 ^       |                                                 |
+                 |       ^   emit .pause();        +=================+     |
+                 |       ^---^---------------------+  return false;  <-----+---+
+                 |                                 +=================+         |
+                 |                                                             |
+                 ^   when queue is empty   +============+                      |
+                 ^---^-----------------^---<  Buffering |                      |
+                     |                     |============|                      |
+                     +> emit .drain();     |  <Buffer>  |                      |
+                     +> emit .resume();    +------------+                      |
+                                           |  <Buffer>  |                      |
+                                           +------------+  add chunk to queue  |
+                                           |            <--^-------------------<
+                                           +============+
+```
+
+
+_Note_: The `.pipe()` function is typically where backpressure is invoked. In 
+an isolate application, both [`Readable`][] and [`Writable`][] streams 
+should be present. If you are writing an application meant to accept a 
+[`Readable`][] stream, or pipes to a [`Writable`][] stream from another app, 
+you may omit this detail.
```

Comment:
Not sure what "application" / "app" means here. A standalone program? A library? A function?